regolith/helpers/u_milestonehelper.py

```diff
@@ -110,161 +113,177 @@ def construct_global_ctx(self):
-                if rc.current:
-                    if j.get("status") not in PROJECTUM_ACTIVE_STATI:
-                        continue
-                print(
-                    f"{i}. {j.get('name')}    due date: {j.get('due_date')}"
-                    f"    status: {j.get('status')}")
-                if rc.verbose:
-                    if j.get("audience"):
-                        print(f"     audience: {j.get('audience')}")
-                    if j.get("objective"):
-                        print(f"     objective: {j.get('objective')}")
-                    if j.get("type"):
-                        print(f"     type: {j.get('type')}")
-                    if j.get('notes'):
-                        print(f"     notes:")
-                        for note in j.get('notes', []):
-                            print(f"       - {note}")
-                if j.get('identifier') == 'deliverable':
-                    del j['name']
-            return
-        rc.index = rc.index.replace(" ", "")
-        if "-" in rc.index:
-            idx_parsed = [i for i in range(int(rc.index.split('-')[0]),
-                                           int(rc.index.split('-')[1]) + 1)]
-        elif "," in rc.index:
-            idx_parsed = [int(i) for i in rc.index.split(',')]
-        else:
-            idx_parsed = [int(rc.index)]
         new_mil = []
-        for idx in idx_parsed:
-            pdoc = {}
-            if idx == 1:
-                mil = {}
-                if not rc.due_date or not rc.name or not rc.objective:
-                    raise RuntimeError(
-                        "name, objective, and due date are required for a new milestone")
-                mil.update({'due_date': rc.due_date})
-                mil['due_date'] = get_due_date(mil)
-                mil.update({'objective': rc.objective, 'name': rc.name, 'uuid': get_uuid()})
-                if rc.audience:
-                    mil.update({'audience': rc.audience})
-                else:
-                    mil.update({'audience': ['lead', 'pi', 'group_members']})
-                if rc.type:
-                    mil.update({'type': rc.type})
-                else:
-                    mil.update({'type': 'meeting'})
-                if rc.status:
-                    mil.update({'status': rc.status})
-                else:
-                    mil.update({'status': 'proposed'})
-                if rc.notes:
-                    mil.update({'notes': rc.notes})
-                mil.update({'identifier': "milestones"})
-                new_mil.append(mil)
-                pdoc = {'milestones': mil}
-            if idx > 1:
-                doc = deepcopy(all_milestones[idx - 2])
-                identifier = doc['identifier']
-                if not doc.get(
-                        'type') and not rc.type and identifier == 'milestones':
+        pdoc = {}
+        all_milestones = []
+        if rc.projectum_id and rc.milestone_uuid:
```

Comment:
we will need tests for these if we don't have them.  put them into a separate test function called `test_helpers_bad()`.  You can mostly copy and paste from the existing test_helpers to get everything set up, but it won't need too much in teh way of outputs so some of the stuff can be deleted.  I can help if you need it. 

You usse this construction to to test that errors are getting raised properly:
```
        with pytest.raises(RuntimeError) as excinfo:
            actual = collect_appts(people, filter_key=key, filter_value=value, begin_date=start, end_date=end)
        assert str(excinfo.value) == expected
```
This checks that RuntimeError is raised and that the error message is as expected.   Then you can test bad inputs using `pytest.mark.parametrize` similar to the successful test.

Reply:
I set up a meeting for tomorrow.  We can discuss.